backend/src/Designer/Services/Implementation/TextsService.cs

```diff
@@ -325,20 +345,89 @@ private async Task UpdateKeysInLayoutsInLayoutSet(string org, string app, string
+                JsonNode originalLayout = layout.DeepClone();
+                foreach (JsonNode layoutObject in layoutArray)
                 {
-                    foreach (TextIdMutation mutation in keyMutations.Where(_ => layoutObject["textResourceBindings"] is not null))
+                    foreach (TextIdMutation mutation in keyMutations)
                     {
-                        layoutObject["textResourceBindings"] = UpdateKey(layoutObject["textResourceBindings"], mutation);
+                        UpdateKeyInLayoutObject(layoutObject, mutation);
                     }
                 }
-                await altinnAppGitRepository.SaveLayout(layoutSetName, layoutName, layout);
+                if (!JsonNode.DeepEquals(layout, originalLayout))
+                {
+                    await altinnAppGitRepository.SaveLayout(layoutSetName, layoutName, layout);
+                }
+            }
+        }
+
+        private static void UpdateKeyInLayoutObject(JsonNode layoutObject, TextIdMutation mutation)
+        {
+            if (layoutObject["textResourceBindings"] is JsonObject)
+            {
+                layoutObject["textResourceBindings"] = UpdateTextResourceKeys(layoutObject["textResourceBindings"], mutation);
+            }
+
+            if (layoutObject["options"] is JsonArray optionsArray)
+            {
+                var options = optionsArray.Deserialize<List<Option>>();
+                if (options != null)
+                {
+                    UpdateOptionListKeys(options, mutation);
+                    layoutObject["options"] = JsonSerializer.SerializeToNode(options);
+                }
+            }
+
+            if (layoutObject["source"] is JsonObject)
+            {
+                UpdateSourceKeys(layoutObject["source"], mutation);
+            }
+        }
+
+        private static void UpdateSourceKeys(JsonNode source, TextIdMutation mutation)
+        {
+            JsonElement jsonElement = source["label"].AsValue().GetValue<JsonElement>();
+            if (jsonElement.ValueKind == JsonValueKind.String && jsonElement.GetString() == mutation.OldId && mutation.NewId.HasValue)
+            {
+                source["label"] = mutation.NewId.Value;
+            }
+        }
+
+        private async Task UpdateKeysInOptionLists(string org, string app, string developer, List<TextIdMutation> keyMutations)
+        {
+            string[] optionListIds = _optionsService.GetOptionsListIds(org, app, developer);
+            foreach (string optionListId in optionListIds)
+            {
+                List<Option> options = await _optionsService.GetOptionsList(org, app, developer, optionListId);
+                foreach (TextIdMutation mutation in keyMutations)
+                {
+                    UpdateOptionListKeys(options, mutation);
+                }
+                await _optionsService.CreateOrOverwriteOptionsList(org, app, developer, optionListId, options);
```

Comment:
It looks like this code also rewrites every option file, even those that haven't been changed.
I suggest optimizing this by only performing read/write operations on the files that are impacted. This would reduce unnecessary operations, as file I/O tends to be slow and resource-intensive.